ethstats/ethstats.go

```diff
@@ -284,65 +302,76 @@ func (s *Service) readLoop(conn *websocket.Conn) {
 
 	for {
 		// Retrieve the next generic network packet and bail out on error
-		var msg map[string][]interface{}
+		var msg interface{}
 		if err := websocket.JSON.Receive(conn, &msg); err != nil {
 			log.Warn("Failed to decode stats server message", "err", err)
 			return
 		}
-		log.Trace("Received message from stats server", "msg", msg)
-		if len(msg["emit"]) == 0 {
-			log.Warn("Stats server sent non-broadcast", "msg", msg)
-			return
-		}
-		command, ok := msg["emit"][0].(string)
-		if !ok {
-			log.Warn("Invalid stats server message type", "type", msg["emit"][0])
-			return
-		}
-		// If the message is a ping reply, deliver (someone must be listening!)
-		if len(msg["emit"]) == 2 && command == "node-pong" {
-			select {
-			case s.pongCh <- struct{}{}:
-				// Pong delivered, continue listening
-				continue
-			default:
-				// Ping routine dead, abort
-				log.Warn("Stats server pinger seems to have died")
+
+		switch packet := msg.(type) {
+		case map[string]interface{}:
+			msgEmit, _ := packet["emit"].([]interface{})
```

Comment:
Also we should check err here and everywhere else that they're ignored with an underscore

Reply:
yeah, that seems like a bigger thing. For now I think it's ok (many of those come from upstream).